client/state/schema-utils.js

```diff
@@ -0,0 +1,188 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import validator from 'is-my-json-valid';
+import { forEach, isEmpty, get, noop } from 'lodash';
+
+/**
+ * Throw an error if not in production and schema is not valid according to spec.
+ *
+ * This is intended to catch invalid schemas early in development rather than shipping
+ * hard to track down bugs.
+ *
+ * @param {Object} schema Schema to validate against the draft spec schema
+ * @throws An error if the schema is invalid according to spec
+ */
+export const throwIfSchemaInvalid = ( () => {
+	if ( process.env.NODE_ENV === 'production' ) {
+		return noop;
+	}
+
+	// JSON Schema meta-schema draft-04 from http://json-schema.org/draft-04/schema
+	// Used to validate a JSON schema schema
+	// The schema is inlined here so it can be easily pruned from production builds.
+	const jsonSchemaSchema = {
```

Comment:
> // The schema is inlined here so it can be easily pruned from production builds.

There are other ways to do this, which may be preferable to including this big chunk inline.  For example:  https://github.com/Automattic/wp-calypso/blob/91b73692/webpack.config.js#L260